algebra/src/groups/linear_combination.rs

```diff
@@ -0,0 +1,67 @@
+use super::Group;
+use crate::fields::Field;
+
+/// Generic struct of a formal linear combination
+pub struct LinearCombination<'a, G: Group>
```

Comment:
I think we can do a lot better here and avoid debatable implementations of Group trait just for the sake of using this.
We can use Rust std::ops traits and define something like

```
use num_traits::{Zero, One};
use std::ops::{AddAssign, Mul, MulAssign};

pub struct LinearCombination<'a, C: MulAssign<C>, E: AddAssign<E> + Mul<C>> {
   item: Vec<(C, &'a E)>
}
```

We don't need more, plus we can use it for any pair (C, E) that supports such operations as it is

Reply:
Done